### PR TITLE
orca: add otool_external hook for external optimizer

### PR DESCRIPTION
**Pull request by markuskowa**
As described in section 9.24.6 in the ORCA manual, ORCA can use an external program for geometry optimization.
However, this script needs to be right next to the ORCA executables. 

We provide two options now:
1. A simple wrapper that forwards the command to a script set in $ORCA_OTOOL_EXTERNAL (default)
2. Fully declarative: provide a path to a script via inputs (requires orca rebuild).

### CODE REVIEW

**Comment by markuskowa** [comment]
> Hm, I can only find these options in chapter 7.26.6. There is not even chapter 9 in the ORCA manual, no?
> https://www.faccts.de/docs/orca/6.0/manual/contents/detailed/geomopt.html#orca-as-external-optimizer

Oops old version 5..., but yes.

**Comment by markuskowa** [comment]
I removed the wrapper script. This may have made sense for orca-5, but since you can specify it in the input or via $EXTOPTEXE in orca-6, this would be redundant.

File: pkgs/by-name/orca/package.nix [hunk 1]
```diff
@@ -7,6 +7,10 @@
 , openssh
 , xtb
 , enableAvx2 ? true
+
+# Provide path to a custom script for otool_external
+, withCustomOtoolExternal ? false
+, customOtoolExternal ? null
```
Comment (sheepforce) [inline pkgs/by-name/orca/package.nix:13]: Should there maybe be an assert that `customOtoolExternal` is not null, if `withCustomOtoolExternal` is set?

Comment (markuskowa) [reply pkgs/by-name/orca/package.nix:13]: ok, now it should fail gracefully.

**Review by sheepforce — APPROVED** [review]

**Pull request merged**